### PR TITLE
feature: added open api merger

### PR DESCRIPTION
**Pull request by adrrf**
This pull request introduces functionality to merge OpenAPI specifications and serve the merged documentation via a Swagger UI. The changes include new dependencies, configuration updates, and the addition of several new functions and modules to handle the OpenAPI merging and serving process.

### Dependencies and Configuration:

* Added `openapiv3` and `walkdir` dependencies to `Cargo.toml`.
* Updated `config.yaml.example` to include `docs_path` and `openapi_path` for OpenAPI documentation paths.

### New Modules and Functions:

* Added `openapi` module to `src/config/mod.rs` to handle OpenAPI-related functionality.
* Introduced `OpenApiMerger` struct in `src/config/openapi.rs` for merging OpenAPI specifications and generating Swagger UI.
* Updated `src/config/parser.rs` to include `docs_path` and `openapi_path` in `GatewayConfig` struct.

### Main Application Changes:

* Imported `OpenApiMerger` and added `merge_openapi_specs` function in `src/main.rs` to merge specs and generate Swagger UI. [[1]](diffhunk://#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR4-R5) [[2]](diffhunk://#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR31-R32) [[3]](diffhunk://#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR57-R78)
* Added routes in `handle_request` function to serve OpenAPI spec and Swagger UI.

### HTML Template:

* Added `openapi.html` template for serving Swagger UI.

### CODE REVIEW

File: openapi.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            <title>Swagger UI</title>
+            <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/swagger-ui/4.14.0/swagger-ui.css" />
+            <script src="https://cdnjs.cloudflare.com/ajax/libs/swagger-ui/4.14.0/swagger-ui-bundle.js"></script>
+            <script src="https://cdnjs.cloudflare.com/ajax/libs/swagger-ui/4.14.0/swagger-ui-standalone-preset.js"></script>
+          </head>
+          <body>
+            <div id="swagger-ui"></div>
+            <script>
+              const ui = SwaggerUIBundle({
+                url: 'http://0.0.0.0:8080/doc/openapi.yaml',  // Use self.url to dynamically insert the URL
```
Comment (TunixR) [inline openapi.html]: get from config

File: src/config/openapi.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                    .unwrap()
+                    .to_str()
+                    .unwrap()
+                    .to_string();
+                self.specs.insert(service_name, spec);
+            }
+        }
+        Ok(())
+    }
+
```
Comment (TunixR) [inline src/config/openapi.rs]: Im not sure if Result<x,y> requires you to also define an Err() return value. plz check

File: src/main.rs [hunk 3]
```diff
@@ -74,6 +86,13 @@ async fn handle_request(
     request_id: String,
 ) -> Result<Response<BoxBody>, GenericError> {
     let path = req.uri().path();
+
+    match path {
+        "/doc/openapi.yaml" => return serve_openapi_spec(&config.openapi_path).await,
+        "/doc" => return serve_swagger_ui(&config.openapi_path).await,
```
Comment (TunixR) [inline src/main.rs]: change to docs

File: src/config/openapi.rs (hunk 2 above)
Comment (adrrf) [reply src/config/openapi.rs]: ? operator automatically propagates the error

**Pull request merged**